### PR TITLE
cleanup(rules): assign `maturity_deprecated` to some rules + improve desc (2/n)

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind feature

> /kind bug

/kind cleanup

> /kind design

/kind documentation

> /kind failing-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area rules

> /area registry

> /area build

> /area documentation

**Proposed rule [maturity level](https://github.com/falcosecurity/rules/blob/main/CONTRIBUTING.md#maturity-levels)**

> Uncomment one (or more) `/area <>` lines (only for PRs that add or modify rules):

/area maturity-stable

/area maturity-incubating

/area maturity-sandbox

> /area maturity-deprecated

**What this PR does / why we need it**:
Continue enhancing the `desc` of the rules, continuation of https://github.com/falcosecurity/rules/pull/141

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -3463,10 +3563,12 @@
 # Rule for detecting potential Log4Shell (CVE-2021-44228) exploitation
 # Note: Not compatible with Java 17+, which uses read() syscalls
 - macro: java_network_read
-  condition: (evt.type=recvfrom and fd.type in (ipv4, ipv6) and proc.name=java)
+  condition: (evt.type=recvfrom and fd.type in (ipv4, ipv6) and proc.exe endswith java)
```
Comment (incertum) [inline rules/falco_rules.yaml:3566]: This cleanup was previously overlooked.

File: rules/falco_rules.yaml [hunk 2]
```diff
@@ -3150,7 +3222,13 @@
 # # Change rule to different port, then different process name, and test again that it fires.
 
 - rule: Outbound or Inbound Traffic not to Authorized Server Process and Port
-  desc: Detect traffic that is not to authorized server process and port.
+  desc: >
```
Comment (incertum) [inline rules/falco_rules.yaml:3225]: We have about 3 sandbox detections that based on how they created the logic expression they won't evaluate to true unless populating the lists. Not a big fan of it, we may want to change them to placeholder always_true macros? @darryk10 just so we actually comply with our own new guidelines?

File: rules/falco_rules.yaml [hunk 3]
```diff
@@ -3070,7 +3139,10 @@
   condition: (user_known_k8s_client_container)
 
 - rule: The docker client is executed in a container
```
Comment (incertum) [inline rules/falco_rules.yaml:3141]: This is a legacy rule name and now misleading. We could use this window of opportunity with lots of changes to adjust the rules name? @leogr what are your thoughts?

File: rules/falco_rules.yaml (hunk 2 above)
Comment (incertum) [reply rules/falco_rules.yaml:3225]: @darryk10 and I synced and we came to the conclusion that 4 rules (including this one) actually to do not conform to the criteria for sandbox according to the maturity framework guidelines.

We decide to assign the `maturity_deprecated` label to those 4 rules.

**Review by leogr — APPROVED** [review]

**Pull request merged**